src/system/AdminModule/Controller/AdminInterfaceController.php

```diff
@@ -0,0 +1,773 @@
+    	$caller['_zkType'] = $master_request->attributes->get('_zkType');
+    	$caller['_zkFunc'] = $master_request->attributes->get('_zkFunc');
+    	    	
+    	
+    	$menuLinks          = isset($params['links'])       ? $params['links'] : '';
+    	$menuId             = isset($params['menuid'])      ? $params['menuid'] : '';
+    	$menuClass          = isset($params['menuclass'])   ? $params['menuclass'] : 'navbar navbar-default navbar-modulelinks navbar-modulelinks-main';
+    	$menuItemClass      = isset($params['itemclass'])   ? $params['itemclass'] : '';
+    	$menuItemFirst      = isset($params['first'])       ? $params['first'] : '';
+    	$menuItemLast       = isset($params['last'])        ? $params['last'] : '';
+    	$returnAsArray      = isset($params['returnAsArray']) ? (boolean) $params['returnAsArray'] : false;
+    	
+    	if (empty($menuLinks)) {
+    		if (!isset($params['modname']) || !ModUtil::available($params['modname'])) {
+    			$params['modname'] = ModUtil::getName();
+    		}
+    	
+    		// check our module name
+    		if (!ModUtil::available($params['modname'])) {
+    		//	$view->trigger_error('modulelinks: '.__f("Error! The '%s' module is not available.", DataUtil::formatForDisplay($params['modname'])));
+    	
+    			return false;
+    		}
+    	
+    		$params['type'] = isset($params['type']) ? $params['type'] : 'admin';
+    	
+    		// get the menu links
+    		// try the Core-2.0 way first, then try the legacy way.
+    		$menuLinks = $this->get('zikula.link_container_collector')->getLinks($params['modname'], $params['type']);
+    		if (empty($menuLinks)) {
+    			$menuLinks = ModUtil::apiFunc($params['modname'], $params['type'], 'getLinks', $params);
+    		}
+    	}
+    	
+    	// return if there are no links to print or template has requested to returnAsArray
+    	if ((!$menuLinks) || ($returnAsArray && isset($params['assign']))) {
+    		if (isset($params['assign'])) {
+    			//$view->assign($params['assign'], $menuLinks);
+    		}
+    		return '';
+    	}
+    	
+    	$html = '';
+    	
+    	if (!empty($menuLinks)) {
+    		$html = '<ul';
+    		$html .= !empty($menuId) ? ' id="'.$menuId.'"' : '';
+    		$html .= !empty($menuClass) ? ' class="'.$menuClass.'"' : '';
+    		$html .= '>';
+    	
+    		$i = 1;
+    		$size = count($menuLinks);
+    		foreach ($menuLinks as $menuitem) {
+    			$class = array();
+    			$class[] = $size == 1 ? 'z-ml-single' : '';
+    			$class[] = ($i == 1 && $size > 1) ? $menuItemFirst : '';
+    			$class[] = ($i == $size && $size > 1) ? $menuItemLast : '';
+    			$class[] = !empty($menuItemClass) ? $menuItemClass : '';
+    			$class[] = (isset($menuitem['disabled']) && $menuitem['disabled'] == true) ? 'z-ml-disabled' : '';
+    			$class = trim(implode(' ', $class));
+    			$i++;
+    	
+    			if (System::isLegacyMode() && !empty($class) && isset($menuitem['class'])) {
+    				if ($menuitem['class'] == 'z-icon-es-add') {
```

Comment:
Yes this was copy & paste many thing's still not working I will be doing all issues you will point out so please comment what you think could be solved differently. This is very initial state we will need to work on it. I will update PR with issues and ideas to discuss.